utoipa-swagger-ui/build.rs

```diff
@@ -18,17 +18,7 @@ fn main() {
         SWAGGER_UI_DIST_ZIP
     );
 
-    let target_dir = env::var("CARGO_TARGET_DIR")
-        .or_else(|_| env::var("CARGO_BUILD_TARGET_DIR"))
-        .or_else(|_| -> Result<String, VarError> {
-            let manifest_dir = env::var("CARGO_MANIFEST_DIR")?;
-            if manifest_dir.contains("/package") {
-                Ok("target".to_string())
-            } else {
-                Ok("../target".to_string())
-            }
-        })
-        .unwrap();
+    let target_dir = env::var("OUT_DIR").unwrap();
```

Comment:
The reason for logic behind is that the docs.rs uses custom target dir for the artifacts. And we cannot use `OUT_DIR` for the artifacts because it wont be constant place thus `rust-embed` will break (at least previously). This would need some throughout testing with actual API using dev build and release build to see whether it actually works. 

There is also branch: https://github.com/juhaku/utoipa/tree/test-swagger-build-attr for fixing  the target dir assumption many are facing. But not merged yet.

Reply:
:ok_hand: Good, two things I experienced before was. 
1. docs.rs has custom builder setup which relies on custon target dir location otherwise the build will completely fail there. 
2. When I used `OUT_DIR` the build out dir was not constant thus building once the app and then second time it did not produce output dir resulting empty pages  / or did not find anything to show via RustEmbed. This happened with older Rust versions though.

Reply:
I think this `OUT_DIR` can be tried out. But it need to be released alone as single change and first as a pre-release in case there are some funny outcomes in docs.rs etc. 

I don't have the error reports any more and docs.rs does not keep that long history of builds. And in my understanding the only place to find out incompatibilities is to browse the build logs of docs.rs.